### ISSUE TITLE
Add negative tests

### ISSUE THREAD

**Issue by rennergade** [Lind-Project/lind_project#224]
One problem I've been seeing with our test suite is that they all test for positive outcomes. Let's make some tests that intentionally fail and test to make sure they fail.

**Comment by rennergade** [comment]
@AlpacaMax once you sort out the test suite bug, it would be great for you to add some negative tests to the suite. I'll comment here with some ideas, or feel free to brainstorm some of your own.

**Comment by rennergade** [comment]
I think starting with some [mmap](https://man7.org/linux/man-pages/man2/mmap.2.html) tests would be good. Read the man page and check the error section to get some ideas.

**Comment by rennergade** [comment]
After talking with @justyoyo123 and @AlpacaMax 

@justyoyo123 is going to get started with some negative tests for mmap()/munmap()

and @AlpacaMax is going to add some negative tests for fork() and exec()

A good way to think of tests here is to checkout out the manpages for these syscalls, and see what errors are expected in different failure cases. Then designing the test to purposely hit those error codes.

**Comment by AlpacaMax** [comment]
> After talking with @justyoyo123 and @AlpacaMax
> 
> @justyoyo123 is going to get started with some negative tests for mmap()/munmap()
> 
> and @AlpacaMax is going to add some negative tests for fork() and exec()
> 
> A good way to think of tests here is to checkout out the manpages for these syscalls, and see what errors are expected in different failure cases. Then designing the test to purposely hit those error codes.

Sounds good. Do we just add negative cases as individual files or add them into the files we already have?

**Comment by rennergade** [comment]
Yeah, these should be new testfiles. If you want to test for multiple errors inside these new files and it makes sense to do so, you can.
